test/join.jl

```diff
@@ -1557,4 +1559,27 @@ end
                     c="c", d="d")
 end
 
+@testset "matchmissing :notequal correctness" begin
+    name = DataFrame(ID = Union{Int, Missing}[1, 2, missing],
+                    Name = Union{String, Missing}["John Doe", "Jane Doe", "Joe Blogs"])
+    noid = DataFrame(ID = Union{Int, Missing}[], Name = String[])
+    missid = DataFrame(ID = Union{Int, Missing}[missing, missing, missing],
+                    Name = String["John Doe", "Jane Doe", "Joe Blogs"])
+    job = DataFrame(ID = Union{Int, Missing}[missing, 2, 2, 4],
```

Comment:
maybe for `job` add several options (so that we have a loop both on left and right table)

Reply:
Done